### PR TITLE
chore(ec2): add ListAZs method

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/aws/ec2/ec2_test.go [hunk 1]
```diff
@@ -158,6 +158,93 @@ func TestEC2_ListVPC(t *testing.T) {
...
+							Name:   aws.String("zone-type"),
+							Values: aws.StringSlice([]string{"availability-zone"}),
+						},
+						{
+							Name:   aws.String("state"),
+							Values: aws.StringSlice([]string{"available"}),
+						},
+					},
+				}).Return(&ec2.DescribeAvailabilityZonesOutput{
+					AvailabilityZones: []*ec2.AvailabilityZone{
```
Comment (huanjani) [inline internal/pkg/aws/ec2/ec2_test.go:188]: Should we add one or two that get filtered out?

Comment (efekarakus) [reply internal/pkg/aws/ec2/ec2_test.go:188]: Luckily the filtering doesn't happen client side ✌️ ! EC2 should return to us only AZs that are available so we don't have to do any filtering ourselves

Comment (huanjani) [reply internal/pkg/aws/ec2/ec2_test.go:188]: Ah, gotcha gotcha. Right, makes sense. 👍🏼

**Review by huanjani — APPROVED** [review]
LGTM!

File: Makefile [hunk 2]
```diff
@@ -155,7 +155,7 @@ gen-mocks: tools
 	${GOBIN}/mockgen -package=mocks -destination=./internal/pkg/aws/ec2/mocks/mock_ec2.go -source=./internal/pkg/aws/ec2/ec2.go
 	${GOBIN}/mockgen -package=mocks -destination=./internal/pkg/aws/identity/mocks/mock_identity.go -source=./internal/pkg/aws/identity/identity.go
 	${GOBIN}/mockgen -package=mocks -destination=./internal/pkg/aws/route53/mocks/mock_route53.go -source=./internal/pkg/aws/route53/route53.go
-	${GOBIN}/mockgen -package=mocks -destination=./internal/pkg/aws/route53/mocks/mock_route53_domain.go -source=./internal/pkg/aws/route53/route53domains.go
+	${GOBIN}/mockgen -package=mocks -destination=./internal/pkg/aws/route53/mocks/mock_route53_domain.go -source=./internal/pkg/aws/route53/domain.go
```
Comment (Lou1415926) [inline Makefile:158]: :homer-disappearing:

File: internal/pkg/aws/ec2/ec2.go [hunk 3]
```diff
@@ -183,6 +187,33 @@ func (c *EC2) ListVPCs() ([]VPC, error) {
...
+	resp, err := c.client.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("zone-type"),
+				Values: aws.StringSlice([]string{"availability-zone"}),
+			},
+			{
+				Name:   aws.String("state"),
+				Values: aws.StringSlice([]string{"available"}),
+			},
```
Comment (Lou1415926) [inline internal/pkg/aws/ec2/ec2.go:201]: Would we need the "opt-in-status" filter?

Comment (efekarakus) [reply internal/pkg/aws/ec2/ec2.go:201]: My understanding from the SDK description is that we don't need to: https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeAvailabilityZonesInput
```
    // Include all Availability Zones, Local Zones, and Wavelength Zones regardless
    // of your opt-in status.
    //
    // If you do not use this parameter, the results include only the zones for
    // the Regions where you have chosen the option to opt in.
    AllAvailabilityZones *bool `type:"boolean"`
```
(we don't use the param)

**Review by Lou1415926 — APPROVED** [review]

**Pull request merged**